lib/src/text_controller.dart

```diff
@@ -0,0 +1,78 @@
+part of 'hooks.dart';
+
+class _TextEditingControllerHookCreator {
+  const _TextEditingControllerHookCreator();
+
+  /// Creates a [TextEditingController] that will be disposed automatically.
+  ///
+  /// The [text] parameter can be used to set the initial value of the
+  /// controller.
+  TextEditingController call({String text, List<Object> keys}) {
+    return Hook.use(_TextEditingControllerHook(text, null, keys));
+  }
+
+  /// Creates a [TextEditingController] from the initial [value] that will
+  /// be disposed automatically.
+  TextEditingController fromValue(TextEditingValue value, [List<Object> keys]) {
+    return Hook.use(_TextEditingControllerHook(null, value, keys));
+  }
+}
+
+/// Functions to create a text editing controller, either via an initial
+/// text or an initial [TextEditingValue].
+///
+/// To use a [TextEditingController] with an optional initial text, use
+/// [_TextEditingControllerHookCreator.call]:
```

Comment:
The dartdoc shouldn't point to private class.

We can probably just remove the mention to [_TextEditingControllerHookCreator.call], as there's an example right next to this description.

Reply:
I've removed both references to the private class. This also clears my concerns on callable methods with dartdoc :)